pkg/overlay/overlay.go

```diff
@@ -65,24 +97,31 @@ func generateOverlayStructure(containerDir string, rootUID, rootGID int) (string
 // from the source system.  It then mounts up the source directory on to the
 // generated mount point and returns the mount point to the caller.
 func Mount(contentDir, source, dest string, rootUID, rootGID int, graphOptions []string) (mount specs.Mount, Err error) {
-	return mountHelper(contentDir, source, dest, rootUID, rootGID, graphOptions, false)
+	overlayOpts := Options{GraphOpts: graphOptions, ReadOnly: false, RootUID: rootUID, RootGID: rootGID}
+	return MountWithOptions(contentDir, source, dest, &overlayOpts)
 }
 
 // MountReadOnly creates a subdir of the contentDir based on the source directory
 // from the source system.  It then mounts up the source directory on to the
 // generated mount point and returns the mount point to the caller.  Note that no
 // upper layer will be created rendering it a read-only mount
 func MountReadOnly(contentDir, source, dest string, rootUID, rootGID int, graphOptions []string) (mount specs.Mount, Err error) {
-	return mountHelper(contentDir, source, dest, rootUID, rootGID, graphOptions, true)
+	overlayOpts := Options{GraphOpts: graphOptions, ReadOnly: true, RootUID: rootUID, RootGID: rootGID}
+	return MountWithOptions(contentDir, source, dest, &overlayOpts)
 }
 
+// MountWithOptions creates a subdir of the contentDir based on the source directory
+// from the source system.  It then mounts up the source directory on to the
+// generated mount point and returns the mount point to the caller.
+// But allows api to set custom workdir, upperdir and other overlay options
+// Following API is being used by podman at the moment
 // NOTE: rootUID and rootUID are not yet used.
```

Comment:
(This note can probably be dropped here, it would have to talk about `opts.RootUID`, and that is already documented.)